### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by shcherbin**
This PR Introduces the following changes:

- Build Matrix to test against multiple Python versions;
- Enable linting and code coverage  reports;
- Simplify local development with pipenv;
- Use fixtures from dask.distributed test utils;

I hope it will make the project easier to maintain and bring it closer being to continuously published to PyPi.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,12 +1,18 @@
 [bdist_wheel]
 universal=1
 
-[flake8]
```
Comment (SfinxCZ) [inline setup.cfg]: Why did you remove flake8 config?

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+pylama = "*"
+pandas = {version = ">=0.25.3"}
+dask = {extras = ["dataframe", "array"],version = "*"}
+dask-ml = {version = ">=1.1.1"}
+sparse = {version = "==0.5.0"}
+scipy = {version = ">=1.3.1"}
+scikit-learn = {version = ">=0.21.3"}
+
+[requires]
+python_version = "3.7"
```
Comment (SfinxCZ) [inline Pipfile]: You have discrepancy between setting `python_version` in `Pipfile` and python versions you use in test matrix

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,14 @@
+DOCKER_TAG?=latest
+PYTHON_VERSION?=3.7
+
+init:
```
Comment (SfinxCZ) [inline Makefile]: Why is the `init` rule which is not used anywhere?

File: Dockerfile.test [hunk 4]
```diff
@@ -1,22 +1,8 @@
...
-        libssl-dev \
-        automake \
-        libtool \
-        && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/*
-
-ADD . /app
-ENV PYTHONPATH=/app:/usr/lib/python3.6/site-packages/
+ARG version
```
Comment (SfinxCZ) [inline Dockerfile.test]: I usually use uppercase naming convention for build arguments to better separate these variables from the rest of the code.

File: setup.cfg (hunk 1 above)
Comment (shcherbin) [reply setup.cfg]: flake8 was not listed in the dev deps in setup.py This statement has no effect. 
The linting is performed by pylama and its pytest integration.

File: Makefile (hunk 3 above)
Comment (shcherbin) [reply Makefile]: It's for local devel and can be removed if you prefer not having test/dev only code in the codebase.

File: Dockerfile.test (hunk 4 above)
Comment (shcherbin) [reply Dockerfile.test]: Ok.

File: .dockerignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+.pytest_cache/
```
[the remark this answers is not in the document]
Comment (shcherbin) [reply .dockerignore]: I've added only those folders which bloated the docker context during the build. I guess the could be the same.

File: setup.cfg (hunk 1 above)
Comment (SfinxCZ) [reply setup.cfg]: Ah, ok.

File: Pipfile (hunk 2 above)
Comment (shcherbin) [reply Pipfile]: requires section is optional. Removed.

**Comment by shcherbin** [comment]
@SfinxCZ, personally I am not familiar with `poetry` and got quite used to using `pipenv` for most of my projects. However, your arguments are valid and I do respect you as the founder of this particular project and the project at Cisco where it originated. 

I've initiated this PR with the intention to eventually publish this package on PyPi. 
Let's make a bargain and merge it as is, and I will prepare the next one with `poetry` and release logic, ok?

**Comment by SfinxCZ** [comment]
Ok, fair enough. If you need help with migrating to `poetry`, feel free to reach me.
